File: NodeJS/linkedList.js
const { v4: uuidv4 } = require('uuid');

class Node {
  constructor(value) {
    this.id = uuidv4();
    this.value = value;
    this.next = null;
  }
}

class LinkedList {
  constructor() {
    this.head = null;
  }

  add(value) {
    const newNode = new Node(value);

    if (!this.head) {
      this.head = newNode;
      return newNode;
    }

    let current = this.head;
    while (current.next) {
      current = current.next;
    }

    current.next = newNode;
    return newNode;
  }

  remove(id) {
    if (!this.head) {
      return false;
    }

    if (this.head.id === id) {
      this.head = this.head.next;
      return true;
    }

    let current = this.head;
    while (current.next) {
      if (current.next.id === id) {
        current.next = current.next.next;
        return true;
      }
      current = current.next;
    }

    return false;
  }

  toList() {
    const result = [];
    let current = this.head;

    while (current) {
      result.push({
        id: current.id,
        value: current.value
      });
      current = current.next;
    }

    return result;
  }
}

module.exports = LinkedList;

File: LinkedListApi/Models/LinkedList.cs
using System.Collections.Generic;

namespace LinkedListApi.Models
{
    public class LinkedList
    {
        private LinkedListNode? head;

        public LinkedListNode? Head => head;

        public LinkedListNode Add(string value)
        {
            var newNode = new LinkedListNode(value);

            if (head == null)
            {
                head = newNode;
                return newNode;
            }

            var current = head;
            while (current.Next != null)
            {
                current = current.Next;
            }

            current.Next = newNode;
            return newNode;
        }

        public bool Remove(string id)
        {
            if (head == null)
                return false;

            if (head.Id == id)
            {
                head = head.Next;
                return true;
            }

            var current = head;
            while (current.Next != null)
            {
                if (current.Next.Id == id)
                {
                    current.Next = current.Next.Next;
                    return true;
                }
                current = current.Next;
            }

            return false;
        }

        public List<object> ToList()
        {
            var result = new List<object>();
            var current = head;

            while (current != null)
            {
                result.Add(new { current.Id, current.Value });
                current = current.Next;
            }

            return result;
        }
    }
}